core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/request/io/ImportDefectAcceptor.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2020 B2i Healthcare Pte Ltd, http://b2i.sg
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.b2international.snowowl.core.request.io;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Supplier;
+
+import org.eclipse.core.runtime.ISafeRunnable;
+
+import com.b2international.commons.StringUtils;
+import com.b2international.snowowl.core.request.io.ImportDefect.ImportDefectType;
+import com.google.common.base.CharMatcher;
+import com.google.common.collect.Lists;
+
+/**
+ * @since 7.12 
+ */
+public final class ImportDefectAcceptor {
+
+	private final String file;
+	private final List<ImportDefect> defects = Lists.newArrayList();
+
+	public ImportDefectAcceptor(String file) {
+		this.file = file;
+	}
+	
+	public List<ImportDefect> getDefects() {
+		return defects;
+	}
+	
+	public void error(String message) {
+		new ImportDefectBuilder().error(message);
+	}
+	
+	public void warn(String message) {
+		new ImportDefectBuilder().warn(message);
+	}
+	
+	public void info(String message) {
+		new ImportDefectBuilder().info(message);
+	}
+	
+	public ImportDefectBuilder on(String location) {
+		return new ImportDefectBuilder().on(location);
+	}
+	
+	/**
+	 * @since 7.12
+	 */
+	public final class ImportDefectBuilder {
+		
+		private String location;
+		private Supplier<Boolean> when;
+
+		public ImportDefectBuilder on(String location) {
+			this.location = location;
+			return this;
+		}
+
+		public ImportDefectBuilder whenThrows(ISafeRunnable runnable) {
+			return when(() -> {
+				try {
+					runnable.run();
+					return false;
+				} catch (Exception e) {
+					return true;
+				}
+			});
+		}
+		
+		public ImportDefectBuilder whenEqual(Object a, Object b) {
+			return when(Objects.equals(a, b));
+		}
+		
+		public ImportDefectBuilder whenNotEqual(Object a, Object b) {
+			return when(!Objects.equals(a, b));
+		}
+		
+		public ImportDefectBuilder whenBlank(String value) {
+			return when(StringUtils.isEmpty(value));
+		}
+		
+		public ImportDefectBuilder whenNaN(String value) {
+			return when(!CharMatcher.inRange('0', '9').matchesAllOf(value));
```

Comment:
Are floating point dots/commas and negative signs allowed? Maybe E for powers of 10? 😃

Reply:
Let's keep it simple for the initial release, but this would be great to support.